plugins/UFM_NDT_Plugin/ufm_sim_web_service/merger_resources.py

```diff
@@ -620,6 +623,186 @@ def post(self):
+            # local request - no need credentials
+            result, status_code = get_local_cable_validation_report(self.ufm_port)
+        else:
+            cv_username, cv_password = read_cv_credentials(self.cv_credentials_path)
+            if cv_username and cv_password:
+                result, status_code = get_cable_validation_report(self.cable_validation_server_addr,
+                                           self.cable_validation_request_port,
+                                           cv_username, cv_password)
+            else:
+                return self.report_error(status_code, "Failed to read CV credentials.")
+        if result:
+            return self.report_success(result)
+        else:
+            return self.report_error(status_code, "No cable validation report data was retrieved")
+
+    def post(self):
+        '''
+        Post rest 
+        '''
+        return self.report_error(405, "Method is not allowed")
+
+
+class MergerCableValidationEnabled(UFMResource):
+    '''
+    If cable validation server address set
+    and send respond back
+    '''
+    def __init__(self):
+        super().__init__()
+
+    def get(self):
+        '''
+        Get rest 
+        '''
+        if self.cable_validation_server_addr:
+            return self.report_success()
+        else:
+            return self.report_error(400, "Cable validation is not enabled.")
+
+    def post(self):
+        '''
+        Post rest 
+        '''
+        return self.report_error(405, "Method is not allowed")
+
+class MergerCableValidationUpdCred(UFMResource):
+    '''
+    request to update credentials for cable validation connectivity
+    '''
+    def __init__(self):
+        super().__init__()
+
+    def get(self):
+        return self.report_error(405, "Method is not allowed")
+
+    def parse_request(self, json_data):
+        logging.debug("Parsing JSON request: {}".format(json_data))
+        try:
+            self.expected_keys = ["address", "port","username","password"]
+            response, status_code = self.check_request_keys(json_data)
+            if status_code != self.success:
+                return self.report_error(status_code, response)
+        except TypeError:
+            return self.report_error(400, "Failed to verify connectivity parameters")
+        return self.report_success()
+
+    def post(self):
+        '''
+        Post rest - receive connectivity parameters and update in backend
+        {
+          "address": '<IP | Hostname>',
+          "port": '8633',
+          "username": "username",
+          "password": "password"
+        }
+        '''
+        if request.json:
+            json_data = request.get_json(force=True)
+            response, status_code = self.parse_request(json_data)
+            if status_code != self.success:
+                return response, status_code
+            cv_address = json_data["address"]
+            cv_port = json_data["port"]
```

Comment:
the username, password & port in case of local, won't be inside the payload, they're required only in case of remote

use .get instead of [""] will return None instead of raising exception in case the payload doesn't contain these fields 

```suggestion
            cv_port = json_data.get("port")
            cv_username = json_data.get("username")
            cv_password = json_data.get("password")
```